compiler/transforms/snax_copy_to_dma.py

```diff
@@ -201,15 +219,39 @@ def match_and_rewrite(self, op: CopyOp, rewriter: PatternRewriter):
         ops_to_insert.append(pointer_dst)
 
         # apply offset if it is not zero
-        if tsl_source.data.offset:
-            offset = Constant.from_int_and_width(tsl_source.data.offset, IndexType())
-            pointer_src = Addi(pointer_src, offset, IndexType())
-            ops_to_insert.extend([offset, pointer_src])
-
-        if tsl_dest.data.offset:
-            offset = Constant.from_int_and_width(tsl_dest.data.offset, IndexType())
-            pointer_dst = Addi(pointer_dst, offset, IndexType())
-            ops_to_insert.extend([offset, pointer_dst])
+        if tsl_source.data.offset != 0:
+            # Dynamic offset
+            if tsl_source.data.offset is None:
+                assert isinstance(op.source.type.layout, StridedLayoutAttr)
+                offset_op = ExtractStridedMetaDataOp(op.source)
+                # Calculate number of bytes in type
+                el_bytes = op.source.type.element_type.width.data // 8
+                el_bytes_op = Constant.from_int_and_width(el_bytes, IndexType())
+                calc_offset_op = Muli(el_bytes_op, offset_op.offset ,IndexType())
```

Comment:
```suggestion
                calc_offset_op = Muli(el_bytes_op, offset_op.offset, IndexType())
```
strange that formatting does not catch this one

Reply:
I'm pretty sure it is catching it :smile: